src/System.Windows.Forms/src/System/Windows/Forms/FileDialog.cs

```diff
@@ -404,17 +409,17 @@ public bool ValidateNames
         /// </summary>
         private bool DoFileOk(IntPtr lpOFN)
         {
-            NativeMethods.OPENFILENAME_I ofn = Marshal.PtrToStructure<NativeMethods.OPENFILENAME_I>(lpOFN);
+            NativeMethods.OPENFILENAME_I? ofn = Marshal.PtrToStructure<NativeMethods.OPENFILENAME_I>(lpOFN);
```

Comment:
This can't be null.
Also we should replace `Marshal` with the direct unsafe cast, like we do in other places. Something like
```suggestion
            NativeMethods.OPENFILENAME_I* ofn = (NativeMethods.OPENFILENAME_I*)lpOFN;
```

I'm on a phone so can't give a better suggestion.

Reply:
I am observing a few errors when I do this change.

Reply:
Yes, the [`OPENFILENAMEW` struct](https://docs.microsoft.com/windows/win32/api/commdlg/ns-commdlg-openfilenamew) is currently declared as a class, and it needs to be converted to a blittable equivalent.
This definitely falls beyond the scope of this PR.